src/argtypes.c

```diff
@@ -307,7 +307,7 @@ TypeTuple *toArgTypes(Type *t)
         void visit(TypeStruct *t)
         {
             //printf("TypeStruct::toArgTypes() %s\n", t->toChars());
-            if (!t->sym->isPOD())
+            if (!t->sym->isPOD() || t->sym->fields.dim == 0)
```

Comment:
Looks like it was passed in memory before (2.066.1).
I think it should match the ABI of a struct with a `void*` as first field.
```d
private template __externC(RT, P...)
{
    alias extern(C) RT function(P) nothrow @nogc __externC;
}

void bar()
{
    int a;
    struct S
    {
        int foo() { return a; }
    }
    pragma(msg, S.sizeof);
    __externC!(void, S) func;
    func(S());
}
```